workflow-templates/check-go-task.yml

```diff
@@ -159,8 +123,10 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           version: 3.x
 
-      - name: Install golint
-        run: go install golang.org/x/lint/golint@latest
+      - name: Install golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.54
```

Comment:
I wonder if it would be better to set this to `latest`?

The benefit of pinning to a specific version is it avoids the chance of the CI breaking at any time due to a new release of golangci-lint.

However, experience shows that we will almost never update things like this where there isn't a mechanism for getting automated updates. So pinning the version here means the CI system will most often be using an outdated version of golangci-lint that won't necessarily match the version the contributors have installed on their system (it is unfortunate that Go failed to provide a nice system for managing development tool dependencies equivalent to what we have with pip and npm).

Reply:
This tool relies on a lot of linters that are regularly updated every time we have a minor release of `golangci-lint`. 
The author of `golangci-lint` cannot know upfront if a bump of a linter can cause: a failure or a higher number of false positives.
Due to that in almost every minor release that I've been following for the last 3 years, some linters caused problems and needed to be tweaked in the config. The problem intensifies if:
- a new minor of the go version introduces changes at the language level, like with the generics.
- a lot of linters are used, especially leaving default options untouched.

For this reason, I prefer to pin the golangci-lint version. We can leverage some tools like [updatecli](https://github.com/updatecli/updatecli) or [renovate](https://docs.renovatebot.com/getting-started/use-cases/#custom-dependency-extraction) to be notified of a newer release, and we can bump it (if necessary tweak the config file) without causing any burden to developers that might see their CI failing randomly. 

For the local version of tooling, I agree with you. We have two solutions:
- Check if that specific version matches the one we pinned. 
- use [.tool-versions](https://asdf-vm.com/manage/configuration.html#tool-versions) file